qutebrowser/qt/machinery.py

```diff
@@ -21,66 +33,256 @@
+    default = "default"
+
+    #: The wrapper was faked/patched out (e.g. in tests).
+    fake = "fake"
 
+    #: The reason was not set.
+    unknown = "unknown"
 
-def _autoselect_wrapper():
-    for wrapper in _WRAPPERS:
+
+@dataclasses.dataclass
+class SelectionInfo:
+    """Information about outcomes of importing Qt wrappers."""
+
+    pyqt5: Optional[str] = None
+    pyqt6: Optional[str] = None
+    wrapper: Optional[str] = None
+    reason: SelectionReason = SelectionReason.unknown
+
+    def set_module_error(self, name: str, error: Exception) -> None:
+        """Set the outcome for a module import."""
+        setattr(self, name.lower(), f"{type(error).__name__}: {error}")
+
+    def use_wrapper(self, wrapper: str) -> None:
+        """Set the wrapper to use."""
+        self.wrapper = wrapper
+        setattr(self, wrapper.lower(), "success")
+
+    def __str__(self) -> str:
+        if self.pyqt5 is None and self.pyqt6 is None:
+            # No autoselect -> shorter output
+            return f"Qt wrapper: {self.wrapper} (via {self.reason.value})"
+
+        pyqt6 = self.pyqt6 or "not imported"
+        pyqt5 = self.pyqt5 or "not imported"
+        lines = [
+            "Qt wrapper info:",
+            f"  PyQt6: {pyqt6}",
+            f"  PyQt5: {pyqt5}",
+            f"  -> selected: {self.wrapper} (via {self.reason.value})"
+        ]
+        return "\n".join(lines)
+
+    def to_html(self) -> str:
+        return html.escape(str(self)).replace("\n", "<br>")
+
+
+def _autoselect_wrapper() -> SelectionInfo:
+    """Autoselect a Qt wrapper.
+
+    This goes through all wrappers defined in WRAPPER.
+    The first one which can be imported is returned.
+    """
+    info = SelectionInfo(reason=SelectionReason.auto)
+
+    for wrapper in WRAPPERS:
         try:
             importlib.import_module(wrapper)
-        except ImportError:
-            # FIXME:qt6 show/log this somewhere?
+        except ModuleNotFoundError as e:
+            # Wrapper not available -> try the next one.
+            info.set_module_error(wrapper, e)
             continue
-        return wrapper
+        except ImportError as e:
+            # Any other ImportError -> stop to surface the error.
+            info.set_module_error(wrapper, e)
+            break
+
+        # Wrapper imported successfully -> use it.
+        info.use_wrapper(wrapper)
+        return info
 
-    wrappers = ", ".join(_WRAPPERS)
-    raise Error(f"No Qt wrapper found, tried {wrappers}")
+    # SelectionInfo with wrapper=None but all error reports
+    return info
 
 
-def _select_wrapper():
+def _select_wrapper(args: Optional[argparse.Namespace]) -> SelectionInfo:
+    """Select a Qt wrapper.
+
+    - If --qt-wrapper is given, use that.
+    - Otherwise, if the QUTE_QT_WRAPPER environment variable is set, use that.
+    - Otherwise, use PyQt5 (FIXME:qt6 autoselect).
+    """
+    for name in WRAPPERS:
+        # If any Qt wrapper has been imported before this, all hope is lost.
```

Comment:
Hmm, why do we need this check? I think if we import both wrappers than as long as we don't do anything like create a QApplication with the one we don't want to use then things should keep working fine with the one we do want to use?